lazy_dataset/core.py

```diff
@@ -1850,10 +1849,10 @@ def is_completed(self):
         return len(self.data) >= self.batch_size
 
     def assess(self, example):
-        return not self.is_completed()
+        raise NotImplementedError
 
     def maybe_append(self, example):
-        if self.assess(example):
+        if not self.is_completed() and self.assess(example):
```

Comment:
Can it happen, that `self.is_completed()` is True?
I would expect that `self.is_completed()` has to be False, i.e. use an assert